modules/apm/src/main/java/org/elasticsearch/telemetry/apm/AbstractInstrument.java

```diff
@@ -25,49 +26,51 @@
  * @param <T> delegated instrument
  */
 public abstract class AbstractInstrument<T> implements Instrument {
-    private static final int MAX_NAME_LENGTH = 255;
-    private final AtomicReference<T> delegate;
+    private final AtomicReference<T> delegate = new AtomicReference<>();
     private final String name;
-    private final String description;
-    private final String unit;
+    private final Function<Meter, T> instrumentBuilder;
 
-    @SuppressWarnings("this-escape")
-    public AbstractInstrument(Meter meter, String name, String description, String unit) {
-        this.name = Objects.requireNonNull(name);
-        if (name.length() > MAX_NAME_LENGTH) {
-            throw new IllegalArgumentException(
-                "Instrument name [" + name + "] with length [" + name.length() + "] exceeds maximum length [" + MAX_NAME_LENGTH + "]"
-            );
-        }
-        this.description = Objects.requireNonNull(description);
-        this.unit = Objects.requireNonNull(unit);
-        this.delegate = new AtomicReference<>(doBuildInstrument(meter));
-    }
-
-    private T doBuildInstrument(Meter meter) {
-        return AccessController.doPrivileged((PrivilegedAction<T>) () -> buildInstrument(meter));
+    public AbstractInstrument(Meter meter, Builder<T> builder) {
+        this.name = builder.getName();
+        this.instrumentBuilder = m -> AccessController.doPrivileged((PrivilegedAction<T>) () -> builder.build(m));
```

Comment:
Sorry, I'm not really familiar with the security manager, forgive me my ignorance :/ I'll have to do some homework there! But wondering what change in this PR made it necessary to elevate permissions when constructing the instrument? 

Considering `AbstractInstrument` is part of the exports of the apm module and could be extended, e.g. in a plugin calling arbitrary user code in `builder.build(m)`, is it safe to run that with elevated permissions? What would be the context in such a case?